plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryQueryPageSource.java

```diff
@@ -152,7 +161,21 @@ public long getMemoryUsage()
     public Page getNextPage()
     {
         verify(pageBuilder.isEmpty());
-        for (FieldValueList record : getFutureValue(tableResult).iterateAll()) {
+        TableResult tableResult = getFutureValue(tableResultFuture);
```

Comment:
We can simplify this a bit. We can use future only to execute `client.executeQuery(session, sql).iterateAll().iterator()` and save the iterator from it.
Then we can pull MAX_PAGE_ROW_COUNT records from that iterator directly on main thread per getNextPage call until we run out. I think it's okay if this part is blocking, i'm assuming that bulk of time is spent in executing the query and fetching the results does not take a long time.

Reply:
I disagree. IMO, only referring to the page `MAX_PAGE_ROW_COUNT` in a single place makes the code easier to follow - we set the page size once initially, and then we convert every BigQuery page into a Trino page, one by one.

I'll resolve this, but take a look at the latest code, without the extra futures it might be ok.

Reply:
After looking at it more, I now understand what you meant, and I agree an iterator would work. But IMO the difference is small enough, so I'll leave it as is. LMK if you have strong feelings about it.